lab_3/main.py

```diff
@@ -20,8 +21,41 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
-
+    if not isinstance(text, str):
+        return ()
+    invaluable_trash = ['`', '~', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '-', '+',
+                        '=', '{', '[', ']', '}', '|', '\\', ':', ';', '"', "'", '<', ',', '>',
+                        '/', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
+    for symbol in invaluable_trash:
+        text = text.replace(symbol, '')
+    for key, value in {'ö': 'oe', 'ü': 'ue', 'ä': 'ae', 'ß': 'ss'}.items():
+        text = text.replace(key, value)
+    sentence = ''
+    all_sentences = []
+    for number, symbol in enumerate(text):
+        if symbol not in ['.', '?', '!', '…'] and number + 1 != len(text):
```

Comment:
`['.', '?', '!', '…']` is better to define earlier, outside of the loop.

Reply:
Done